### PR TITLE
libfabric: add 1.18.1

### PR DESCRIPTION
**Pull request by jack-morrison**
Adds yesterday's 1.18.1 from [ofiwg/libfabric](https://github.com/ofiwg/libfabric/releases) release via `spack checksum`.

### CODE REVIEW

**Review by alecbcs — APPROVED** [review]
Confirmed SHA. Thanks for the version bump @jack-morrison!

**Comment by alecbcs** [comment]
Will merge once we get a review from @rajachan and CI completes.

**Review by rajachan — APPROVED** [review]

**Pull request merged**
